tests/18-rdfc-verify.js

```diff
@@ -0,0 +1,172 @@
+     * results.
+     *
+     * Note that a single implementation may define multiple verifiers, which
+     * in turn may support multiple key types.
+     *
+     * We want a test suite for each permutation for every key type we require.
+     */
+
+    for(const [name, {endpoints: verifiers}] of targetImplementations) {
+      for(const verifier of verifiers) {
+        const {
+          supportedEcdsaKeyTypes: verifierKeyTypes
+        } = verifier.settings;
+
+        // Important: iterate over required keytypes!
+        for(const keyType of requiredSupportedKeyTypes) {
+          const supportsKeyType = verifierKeyTypes.includes(keyType);
+
+          if(!supportsKeyType) {
+            console.warn({
+              implementationName: name,
+              requiredKeyType: keyType,
+              verifierKeyTypes
+            }, 'Unsupported keytype for verifier. Skipping test.');
+          }
+
+          const testVector = supportsKeyType ?
+            testVectorsByKeyType.get(keyType) ?? null : null;
+
+          const assertionTitle = `${name}: ${keyType}`;
+          this.implemented.push(assertionTitle);
+
+          // Now we can define all normative statement tests.
+          describe(assertionTitle, function() {
+            itMustVerifyValidVC({
+              implementationName: name,
+              keyType,
+              verifier,
+              testVector,
+            });
+
+            itRejectsInvalidCryptosuite({
+              implementationName: name,
+              keyType,
+              verifier,
+              testVector,
+            });
+          });
+        }
+      }
+    }
+  });
+});
+
+/**
+ * Below are the formal test prototypes for all normative statements
+ * in the specification.
+ */
+
+function itMustVerifyValidVC({
+  implementationName, keyType,
+  verifier, testVector
+}) {
+  return it('MUST verify a valid VC with an ecdsa-rdfc-2019 proof.',
+    async function() {
+      annotateReportableTest(this, {implementationName, keyType});
+
+      if(testVector === null) {
+        // Mark as a failure.
+        throw new Error(
+          `Implementation not marked as supporting required "${keyType}"! ` +
+          'Is keyType missing from `supportedEcdsaKeyTypes`?'
+        );
+      }
+
+      await verificationSuccess({credential: testVector, verifier});
+    });
+}
+
+function itRejectsInvalidCryptosuite({
+  implementationName, keyType,
+  verifier, testVector
+}) {
+  return it('If the "cryptosuite" field is not the string ' +
+            '"ecdsa-rdfc-2019" or "ecdsa-jcs-2019", an error MUST be ' +
+            'raised.',
+  async function() {
+    annotateReportableTest(this, {implementationName, keyType});
+
+    if(testVector === null) {
+      // Mark as a failure.
+      throw new Error(
+        `Implementation not marked as supporting required "${keyType}"! ` +
+          'Is keyType missing from `supportedEcdsaKeyTypes`?'
+      );
+    }
+
```

Comment:
The idea is these would move out of the main runner file and into some separate module that acts as a "repo" for normative statement assertions. A hard reference to the docs could be added to jsdoc for each one, which we could surface in failures via a small change to the reporter.

Reply:
so I was thinking about this, and if we add reportRow we could have report row do something like

```js
function(assertions) {
  //redudant test for before statements
  if(testVector === null) {
    throw new Error(`Implementation ${name} failed to issue Vc.`)
  }
  assertions()
}
```

so we set the title, row, and columnId in one place, do redudant checks, and then run the actual test assertions in the same test.